blob/blob-aws/src/test/java/com/salesforce/multicloudj/blob/aws/AwsBlobBenchmarkTest.java

```diff
@@ -0,0 +1,148 @@
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.http.SdkHttpClient;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.S3Configuration;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+import java.net.URI;
+import java.util.concurrent.ThreadLocalRandom;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.Test;
+import org.openjdk.jmh.runner.RunnerException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class AwsBlobBenchmarkTest extends AbstractBlobBenchmarkTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(AwsBlobBenchmarkTest.class);
+    private static final String endpoint = "https://s3.us-east-2.amazonaws.com";
+    private static final String bucketName = "chameleon-jcloud-benchmarks";
+    private static final String versionedBucketName = "chameleon-jcloud-versioned";
+    private static final String nonExistentBucketName = "java-bucket-does-not-exist";
+    private static final String region = "us-east-2";
+
+    @Override
+    protected Harness createHarness() {
+        return new HarnessImpl();
+    }
+
+    @Override
+    @Test
+    public void runBenchmarks() throws RunnerException {
+        // Configure JMH to run benchmarks and save results to a file
+        Options opt = new OptionsBuilder()
+                .include(".*" + this.getClass().getName() + ".*benchmarkSingleActionPut.*") // Include all benchmark methods in this class
+                .forks(1)
+                .warmupIterations(3)
+                .measurementIterations(5)
+                .resultFormat(org.openjdk.jmh.results.format.ResultFormatType.JSON)
+                .result("target/jmh-results.json") // Use a more specific path
+                .build();
+
+        new Runner(opt).run();
+    }
+
+    public static class HarnessImpl implements Harness {
+        SdkHttpClient httpClient;
+        S3Client client;
+
+        @Override
+        public AbstractBlobStore<?> createBlobStore(boolean useValidBucket, boolean useValidCredentials, boolean useVersionedBucket) {
+            String accessKeyId = System.getenv().getOrDefault("AWS_ACCESS_KEY_ID", "FAKE_ACCESS_KEY");
+            String secretAccessKey = System.getenv().getOrDefault("AWS_SECRET_ACCESS_KEY", "FAKE_SECRET_ACCESS_KEY");
+            String sessionToken = System.getenv().getOrDefault("AWS_SESSION_TOKEN", "FAKE_SESSION_TOKEN");
+
+            if (!useValidCredentials) {
+                accessKeyId = "invalidAccessKey";
+                secretAccessKey = "invalidSecretAccessKey";
+                sessionToken = "invalidSessionToken";
+            }
```

Comment:
This doesn't apply here. We don't need to handle such scenarios. These scenarios are more for functional IT